src/ja/common/docker_context.py

```diff
@@ -65,6 +149,18 @@ def __init__(self, cpu_threads: int = -1, memory: int = 1):
         @param cpu_threads Initial value of @cpu_threads.
         @param memory The value of @memory.
         """
+        self._cpu_threads = -1
+        if cpu_threads != -1:
```

Comment:
Because the setter checks if you are setting the number of CPU threads to a value > 0.
The current behavior is that you are allowed to set cpu_threads to -1 through the constructor, but not through the setter.